core/Enclave/enclave_hsm.cpp

```diff
@@ -294,43 +327,92 @@ sgx_status_t enclave_create_rsa_key(uint8_t *cmk_blob,
         printf("ecall create_rsa_key cmk_keyblob_size:%lu < key_blob_size:%d.\n", cmk_blob_size, real_keyblob_size);
         return SGX_ERROR_INVALID_PARAMETER;
     }
-        rsa_key_data_t *keyblob = (rsa_key_data_t *)cmk_blob;
+    if (keyspec == NULL || padding_mode == NULL)
+        return SGX_ERROR_INVALID_PARAMETER;
+
+    RSA     *rsa            = NULL;
+    BIGNUM  *bne            = NULL;
+    BIO     *public_key     = NULL;
+    BIO     *private_key    = NULL;
+    char    *p_public_key   = NULL;
+    char    *p_private_key  = NULL;
+    string payload;
      
     do {
-        RSA *rsa = RSA_new();
-        BIGNUM *bne = BN_new();
-        BN_set_word(bne, RSA_F4);
-        switch (keyspec)
-        {
+        // init rsa key struct
+        rsa = RSA_new();
+        if (rsa == NULL) {
+            ret = SGX_ERROR_OUT_OF_MEMORY;
+            break;
+        }
+
+        bne = BN_new();
+        if (bne == NULL) {
+            ret = SGX_ERROR_OUT_OF_MEMORY;
+            break;
+        }
+
+        if(!BN_set_word(bne, RSA_F4)) {
+            break;
+        }
+
+        // generate rsa key
+        int res;
```

Comment:
seems it's another common issues in your PR.
"res" is generally for "resource"
"ret" is for the return code, but seems not suitable for your case.

Change it to a valid name!

Reply:
yes, I'll change it